dht/dht.go

```diff
@@ -156,16 +184,18 @@ func NewServer(c *ServerConfig) (s *Server, err error) {
 			panic(err)
 		}
 	}()
-	go func() {
-		err := s.bootstrap()
-		if err != nil {
-			select {
-			case <-s.closed:
-			default:
-				log.Printf("error bootstrapping DHT: %s", err)
+	if !s.config.NoBootstrap {
```

Comment:
Better here to actually move the smarts about NoBootstrap into the bootstrap function. By "turning off" sections of code on special conditions, you create more opportunity for bugs to be missed by creating conditionally dead code. I hope this makes sense.